src/Mealz/MealBundle/Service/ParticipationService.php

```diff
@@ -305,4 +338,11 @@ public function getBookedDishCombination(Profile $profile, Meal $meal): Collecti
 
         return $participant->getCombinedDishes();
     }
+
+    private function isParticipationLocked(Participant $participant): bool
+    {
+        $now = new DateTime('now');
+
+        return $now > $participant->getMeal()->getLockDateTime();
```

Comment:
This is only half of the truth. LockDateTime is in the past then it's true: That's correct. What about the meals that are open (not happened yet)? They are also locked? It's only locked when now is between locked date time and meal lunch time. This check will also return true (locked) for open meals.

Reply:
That't correct. Participations are locked at 4 o'clock afternoon a day before the meal day.

If we consider the participation locked only when now is between locked date time and lunch time, then what will be the status after lunch, i.e. 12 o'clock noon?

Reply:
Maybe I have a different understanding of isLocked.
I guess it should be like:
Now < LockTime < LunchTime: isOpen === true, isLocked === false (You can join, delete, edit combined meals)
LockTime <= Now < LunchTime: isOpen === false, isLocked === true (You can swap, unswap, acceptOffer)
LockTime  < LunchTime <= Now: isOpen === false, isLocked === true (You can't do anything)
Is it correct?
I guess I meant something like: isBookable = isOpen || !isLocked
Now < LockTime < LunchTime: isBookable === true (You can join, delete, edit combined meals)
LockTime <= Now < LunchTime: isBookable === true (You can swap, unswap, acceptOffer)
LockTime  < LunchTime <= Now: isBookable === false (You can't do anything)

Reply:
With this logic we can't prohibit users to book directly between lock-time and lunch time because it'll still be bookable.

Reply:
Yes, correct you need the information that the meal is offered!